rfc-182-flexible-pages.md

```diff
@@ -0,0 +1,176 @@
+
+Option 1: start from a blank page
+
+- users would be able to select from a list of flexible sections and customise them within the constraints defined (e.g. a page title flexible section should always be first)
+
+Option 2: select a preset page type
+
+- users would select a preset type from a list e.g. history page, help page
+- the flexible sections required for that page would be automatically added into the editing window, allowing users to then add content within those flexible sections but not allowing them to reorder them
+
+### Next steps
+
+#### MVP candidate
+
+The page type chosen as the basis for this work is the building history pages, specifically:
+
+- [10 Downing Street](https://www.gov.uk/government/history/10-downing-street)
+- [11 Downing Street](https://www.gov.uk/government/history/11-downing-street)
+- [King Charles Street](https://www.gov.uk/government/history/king-charles-street)
+- [Lancaster House](https://www.gov.uk/government/history/lancaster-house)
+- [1 Horse Guards Road](https://www.gov.uk/government/history/1-horse-guards-road)
+
+There are other pages within the [History of the UK government](https://www.gov.uk/government/history) section, but they have different layouts or are of different page types. They may be considered for work post-MVP.
+
+The chosen pages are relatively simple and their content is currently hard coded in applications, meaning that developers are required to make changes. This MVP will allow us to start with a relatively small number of flexible sections, test many aspects of this work and gain user feedback.
+
+For reference, these are the other candidates that were considered. Some of them may be included in future.
+
+- custom pages for urgent situations (e.g. coronavirus, brexit, cost of living)
+- the GOV.UK roadmap
+- help pages, such as the cookie information page
+- topical events
+
+#### Required flexible sections
+
+To allow the flexible page MVP to recreate history pages, the following will be needed.
+
+Screenshot | Flexible section
+---------- | ----------------
+![Header of the history page](rfc-182/history-pages-1.png) | Standard GOV.UK header, not a flexible section
+![Page title of the history page](rfc-182/history-pages-2.png) | Page title
+![Rich text of the history page](rfc-182/history-pages-3.png) | Rich text with contents list
+
+Page title: content wrapped in a two thirds width column. Content authors would be able to add:
+
+- heading level 1 (required)
+- heading context (optional)
+- lead paragraph (optional)
+
+Rich text with contents list: content wrapped in a one third/two thirds column layout. Content authors would be able to add:
+
+- image src (optional)
+- image alt text (optional)
+- image caption (optional)
+- contents list (ideally automatically generated from content but may not be included for MVP)
+- rich text
```

Comment:
This all looks good to me. 👍 But do you have any thoughts as to how we share the definition of a specific flexible section between publishing and frontend? Are you picturing:

1. some sort of shared 'gem' that publishing and frontend can pull in?
2. some sort of contract / Pact test between Whitehall and frontend?
3. or perhaps a section of Whitehall that the Patterns & Navigation team would 'own' (describing a flexible section's fields & required/optional status etc in YAML)
4. or relying on the two teams being able to have a shared understanding of requirements by talking to one another?

(Same question also applies to "how do we share the definition of a preset layout of flexible sections")?

Reply:
That's a good question, one that I don't think we've thought through yet. I guess I was hoping we can make do with option 4, but maybe we need something more formal/structured?

Reply:
I've added a brief section about this based on our recent conversation, in future considerations.

Reply:
> or perhaps a section of Whitehall that the Patterns & Navigation team would 'own' (describing a flexible section's fields & required/optional status etc in YAML)

This is something that Content publisher experimented with to make adding new document types easier to add. The YAML structure would need very good validation.

Could content-schemas be used to define a flexible section?

Whatever solution is decided on, option 4 should always be a part of it.